src/variant.rs

```diff
@@ -36,16 +37,54 @@
 //! // `get_str` tries to borrow a string slice.
 //! assert_eq!(hello.get_str(), Some("Hello!"));
 //! assert_eq!(num.get_str(), None);
+//!
+//! // Variant carrying a Variant
+//! let variant = Variant::new_variant(&hello);
+//! let variant = variant.get_variant().unwrap();
+//! assert_eq!(variant.get_str(), Some("Hello!"));
+//!
+//! // Variant carrying an array
+//! let array = [Variant::from("Hello"), Variant::from("there!")];
+//! let variant = Variant::new_array::<&str>(&array);
+//! assert_eq!(variant.n_children(), 2);
+//! assert_eq!(variant.get_child_value(0).get_str(), Some("Hello"));
+//! assert_eq!(variant.get_child_value(1).get_str(), Some("there!"));
+//!
+//! // You can also convert from and to a Vec
+//! let array = vec!["Hello", "there!"].to_variant();
+//! assert_eq!(variant.n_children(), 2);
+//! let vec = <Vec<String>>::from_variant(&array).unwrap();
+//! assert_eq!(vec[0], "Hello");
+//!
+//! // Conversion to and from HashMap is also possible
+//! let mut map: HashMap<u16, &str> = HashMap::new();
+//! map.insert(1, "hi");
+//! map.insert(2, "there");
+//! let variant = map.to_variant();
+//! assert_eq!(variant.n_children(), 2);
+//! let map: HashMap<u16, String> = HashMap::from_variant(&variant).unwrap();
+//! assert_eq!(map[&1], "hi");
+//! assert_eq!(map[&2], "there");
+//!
+//! // And conversion to and from tuples.
+//! let variant = ("hello", 42u16, vec![ "there", "you" ],).to_variant();
+//! assert_eq!(variant.n_children(), 3);
+//! assert_eq!(variant.type_().to_str(), "(sqas)");
+//! let tuple = <(String, u16, Vec<String>)>::from_variant(&variant).unwrap();
+//! assert_eq!(tuple.0, "hello");
+//! assert_eq!(tuple.1, 42);
+//! assert_eq!(tuple.2, &[ "there", "you"]);
```

Comment:
Another trivial container that would be useful to add here would be maybes (`m`)

Reply:
ah good point. I will need to add that on zvariant side too and will need it here for testing the serializer against.

Reply:
It wasn't super "trivial" in the end for me but I added good bindings after all. :)

Reply:
I expected it to be basically the same as for arrays/vecs, just with a single item / `Option` :)